elife/postgresql-10.sls

```diff
@@ -0,0 +1,154 @@
+# postgresql-10.sls is intended to be a drop-in replacement for postgresql.sls (psql 9.4)
+# the two are mutually exclusive and share many of the same state names
+
+# copied from postgresql-client.sls
+
+{% set oscodename = salt['grains.get']('oscodename') %}
+
+# http://www.postgresql.org/download/linux/ubuntu/
+postgresql-deb:
+    pkgrepo.managed:
+        # http://www.postgresql.org/download/linux/ubuntu/
+        - humanname: Official Postgresql Ubuntu LTS
+        - name: deb http://apt.postgresql.org/pub/repos/apt/ {{ oscodename }}-pgdg main
+        - key_url: https://www.postgresql.org/media/keys/ACCC4CF8.asc
+
+# copied from 
+
+pgpass-file:
+    file.managed:
+        - name: /root/.pgpass
+        - source: salt://elife/config/root-pgpass
+        - template: jinja
+        - mode: 0600
+        - defaults:
+            user: {{ pillar.elife.db_root.username }}
+            pass: {{ pillar.elife.db_root.password }}
+            host: localhost
+            port: 5432
+
+{% if salt['elife.cfg']('cfn.outputs.RDSHost') %}
+pgpass-rds-entry:
+    file.append:
+        - name: /root/.pgpass
+        - source: salt://elife/config/root-pgpass
+        - template: jinja
+        - defaults:
+            user: {{ pillar.elife.db_root.username }}
+            pass: {{ salt['elife.cfg']('project.rds_password') }}
+            host: {{ salt['elife.cfg']('cfn.outputs.RDSHost') }}
+            port: {{ salt['elife.cfg']('cfn.outputs.RDSPort') }}
+        - require:
+            - pgpass-file
+{% endif %}
+
+postgresql:
+    pkg.installed:
+        - pkgs:
+            - postgresql-10
+            - libpq-dev # headers for building the libraries to them
+        - require:
+            - pkgrepo: postgresql-deb
+
+{% if not salt['elife.cfg']('cfn.outputs.RDSHost') %}
+    service.running:
+        - enable: True
+        - require:
+            - pkg: postgresql
+            - pgpass-file
+{% else %}
+    service.dead:
+        - enable: False
+        - require:
+            - pkg: postgresql
+            - pgpass-file
+{% endif %}
+
+{% if not salt['elife.cfg']('cfn.outputs.RDSHost') %}
+# 12.04/14.04 legacy file, to be removed with postgresql.sls (psql 9.4)
+# systemd service/target is being used in 16.04+
+postgresql-init:
+    file.absent:
+        - name: /etc/init.d/postgresql
+#        - source: salt://elife/config/etc-init.d-postgresql # remove with postgresql.sls
+{% endif %}
+
+
+# runs pg_upgrade on 9.4 data and then purge postgresql-9.4 
+psql-9.4 to psql-10 migration:
+    file.managed:
+        - name: /root/upgrade-postgresql-9.4-to-10.sh
+        - source: salt://elife/scripts/root-upgrade-postgresql-9.4-to-10.sh
+
+    cmd.script:
```

Comment:
migrate even if using RDS? I think so

Reply:
Depends what the script does. But if you need to change the RDS version you should go through `update_infrastructure` rather than relying on Salt. In general, we can't have a stack with RDS but without EC2 for this reason (Salt doesn't run to complete the setup) so would avoid the coupling.

Reply:
This script relies on `systemctl` and `/var/lib/postgresql/`, can't work with RDS anyway?

Reply:
sorry, I was unclear. Upgrading RDS will happen through `update_infrastructure`, I was just wondering if psql 9.4 data should be upgraded to psql 10 even if local psql isn't being used. I think for consistency it should be. It's just an empty database most likely.

Reply:
Agreed